### PR TITLE
feat: use-rect package

### PR DESCRIPTION
**Pull request by ramzzo**

### CODE REVIEW

**Comment by dammy001** [comment]
@ramzzo lock file is not up to date. Can you run pnpm i locally and push again

**Comment by productdevbook** [comment]
`packages/core/use-composable/src` this code should be here. Let's not pack.

**Comment by dammy001** [comment]
> `packages/core/use-composable/src` this code should be here. Let's not pack.
> 
> 
> 
> 

Yeah, true

File: packages/components/use-rect/src/use-rect.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { onBeforeUnmount, onMounted, ref } from 'vue'
+import { observeElementRect } from '@oku-ui/utils'
+import type { Measurable } from '@oku-ui/utils'
+
+const measurableElement = ref<Measurable | null>(null)
+const rect = ref<DOMRectReadOnly | null>(null)
```
Comment (productdevbook) [inline packages/components/use-rect/src/use-rect.ts:6]: move them into func

**Comment by dammy001** [comment]
@ramzzo can you add minimal test for this?

**Comment by ramzzo** [comment]
> @ramzzo can you add minimal test for this?

I wrote the test but I get an error like this:

```
packages/core/use-composable/tests/use-rect.test.ts [ packages/core/use-composable/tests/use-rect.test.ts ]
Error: Failed to resolve entry for package "@oku-ui/utils". The package may have incorrect main/module/exports specified in its package.json.
```

**Comment by ramzzo** [comment]
Even though I downloaded the oku-ui utils package, I get the error that the oku-ui utils package does not exist. so I can't run the test and see the result. I tried deleting and re-downloading the node_modules and lock file but the situation was the same.

image

image

image

**Comment by productdevbook** [comment]
vitest settings resolve add package, after working in test

**Pull request merged**